### ISSUE TITLE
Home Assistant Error with Tempest Unit

### ISSUE THREAD

**Issue by ddellaterza** [briis/smartweatherudp#16]
I just received a Tempest weather station, and came across your repository. I have installed it using the HACS installation method, and configured the following configuration in my configuration.yaml file. I am currently getting this error when restarting Home Assistant, and the integration doesn't seem to be loading. One subject of note is that I have the Tempest in a separate VLAN, but I allowed traffic between home assistant and this IoT VLAN. I have verified connectivity via ICMP, and the device responds. This is probably something dumb that I'm missing, but I followed the documentation for installing in HACS and I'm not sure what the error is indicating. 

HACS version 1.1.1
HA version 0.111.4

ERROR
```
Error while setting up smartweatherudp platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 178, in _async_setup_platform
    await asyncio.wait_for(asyncio.shield(task), SLOW_SETUP_MAX_WAIT)
  File "/usr/local/lib/python3.7/asyncio/tasks.py", line 442, in wait_for
    return fut.result()
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/smartweatherudp/sensor.py", line 71, in setup_platform
    module = SWReceiver(config.get(CONF_HOST), DEFAULT_PORT, unit_system)
  File "/usr/local/lib/python3.7/site-packages/pysmartweatherudp/receiver.py", line 35, in __init__
    self._socket.bind((host, port))
OSError: [Errno 99] Address not available
```

CONFIGURATION 
```
sensor:
  - platform: smartweatherudp
    host: 192.168.99.44
    wind_unit: mph
    monitored_conditions:
      - temperature
      - dewpoint
      - feels_like
      - heat_index
      - wind_chill
      - wind_speed
      - wind_bearing
      - wind_speed_rapid
      - wind_bearing_rapid
      - wind_gust
      - wind_lull
      - wind_direction
      - precipitation
      - precipitation_rate
      - humidity
      - pressure
      - uv
      - solar_radiation
      - illuminance
      - lightning_count
      - airbattery
      - skybattery
```

**Comment by LuckyJayTx** [comment]
I am thinking that since the hub basically broadcasts the messages via UDP which is connectionless, there is no way for your router to know that those messages need to traverse the VLAN's - in other words, the UDP messages are only available to devices on the same VLAN as the hub.  

Not knowing your network or configuration - is it possible to set up forwarding of the UDP port (50222) from the VLAN the hub is on to the VLAN your Home Assistant is on?

That's my quick thought on your issue...

mark

**Comment by brentpabst** [comment]
Having the same issue here.  My network is just a simple Google Wifi network with open access to all nodes, so there isn't any firewall blocking things or different VLANs in the way here.  

Running the following:
HassOS 4.10
Supervisor 227
HA 111.4

**Comment by ddellaterza** [comment]
I'm suspecting the issue might be with Docker. Not sure if @brentpabst is also using the docker supervised setup or not. I was able to do a tcpdump on the docker server, and I can see the traffic coming from the Tempest device, which is broadcast traffic on port 50222 as @LuckyJayTx mentioned. So the traffic does hit my docker server. Not sure if anyone has gotten this working in a docker environment or not. 

I'm trying to find more info, but either the add-on is having issues attaching to the interface, or the broadcast traffic does not reach the container. I suspect the former, since the container is on the host network.  I think the VLAN is ruled out at this point, since I see the broadcast UDP traffic at the docker host.

![image](https://user-images.githubusercontent.com/36140029/85637279-079b4580-b640-11ea-8400-29d49b5b930f.png)

**Comment by brentpabst** [comment]
Yep, I'm running Docker on an rpi.  Imagine you're right that the container isn't getting access to UDP data from the docker network it's assigned to.  My next step was to try and connect to my rpi and check out the docker networking configs that come with hass.io

**Comment by brentpabst** [comment]
Found a few links that point to the fact that Docker doesn't pass multicast UDP traffic to containers.  One possible option is to run in host networking mode but that seems a bit clumsy.  Also not sure how to modify hass.io to run in host mode yet either.  I'm a bit surprised other UDP based add-ons aren't running into this issue.  Wondering if they're using MQTT etc which gets around some of this?

https://community.home-assistant.io/t/which-dcoker-network-mode/196058
https://stackoverflow.com/questions/28738647/forward-udp-multicast-from-eth0-to-docker0

**Comment by LuckyJayTx** [comment]
well - I am running Home Assistant in Docker (on an unRaid server) - it is indeed running in host network.  
the other difference is my configuration.yaml:

sensor:
  - platform: smartweatherudp
     name: tempest

I'm letting the host default to 0.0.0.0 - I don't think I could get it to work at first with a specific IP address and removed the host parameter.  That might help?  Don't know?

**Comment by brentpabst** [comment]
AHHH! @LuckyJayTx removing the host setting I had specified fixed it for me.  Just started chugging away.  Weird!

**Comment by ddellaterza** [comment]
@LuckyJayTx That looks like it did the trick, removing the host parameter. I figured I would need it because of the VLAN setup, but I believe I was misunderstanding the function of that configuration. I set it to 0.0.0.0 and it built all of the entities! Thansk for the information, and also thanks for having this integration made, I'm super excited to integrate weather conditions into my home, and your work to do this is immensely appreciated! I will close this out!

![image](https://user-images.githubusercontent.com/36140029/85641026-a5940d80-b64a-11ea-8b58-028a8c52db6e.png)
